ddtrace/internal/flare.py

```diff
@@ -29,13 +28,33 @@
 log = get_logger(__name__)
 
 
+@dataclasses.dataclass
+class FlarePrepRequest:
+    log_level: int
+
+    def __init__(self, log_level_name: str):
+        log_level: int | str = logging.getLevelName(log_level_name)
+        if type(log_level) == int:
+            self.log_level = log_level
+        else:
+            raise TypeError("Invalid log level provided: %s", log_level)
```

Comment:
Rather than overwriting the `__init__` it could be better to add a classmethod that creates a proper `FlarePrepRequest` from a log level name instead. Otherwise the typing/autocomplete for the dataclass might get weird? e.g. it only has an int but only takes a str (maybe just my own biased opinion?)


```suggestion
    @classmethod
    def from_log_level_name(cls, log_level_name: str) -> 'FlarePrepRequest':
        log_level: int | str = logging.getLevelName(log_level_name)
        if type(log_level) == int:
            return cls(log_level)
        else:
            raise TypeError("Invalid log level provided: %s", log_level)
```

Reply:
Hmm I think I'm a bit confused about the typing concern? Are you saying we won't know whether we have a string or int log level to pass into the `FlarePrepRequest` init? I don't think that should be a concern given the structure of the `AGENT_CONFIG` product (for now).

Reply:
Chatted offline. Decided this was probably unnecessary, and we can just pass in the str form and let the class method do the conversions.